specs/eventing/data-plane.md

```diff
@@ -1,125 +1,167 @@
+  programming languages leveraging existing library investments in HTTP.
+
+- Knative Eventing assumes a sender-driven (push) event delivery system. That
+  is, each event processor is actively responsible for an event until it is
+  handled (or affirmatively delivered to all following recipients).
+  
+- Knative Eventing aims to make writing [event
+  sources](./overview.md#event-source) and event-processing software easier to
+  write; as such, it imposes higher standards on system components like
+  [brokers](./overview.md#broker) and [channels](./overview.md#channel) than on
+  edge components.
+
+This contract defines a mechanism for a single event sender to reliably deliver
+a single event to a single recipient. Building from this primitive, chains of
+reliable event delivery and event-driven applications can be built.
+
+## Background
 
 The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
 "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
 interpreted as described in RFC2119.
 
-## Data plane contract for Sinks
+When not specified in this document, the [CloudEvents HTTP bindings, version
+1](https://github.com/cloudevents/spec/blob/v1.0.1/http-protocol-binding.md) and
+[HTTP 1.1 protocol](https://tools.ietf.org/html/rfc7230) standards should be
+followed (with the CloudEvents bindings preferred in the case of conflict).
 
-A **Sink** MUST be able to handle duplicate events.
+The current version of this document does not describe protocol negotiation or
+the ability to upgrade an HTTP 1.1 event delivery into a more efficient protocol
+such as GRPC, AMQP or the like. It is expected that a future compatible version
+of this specification might describe a protocol negotiation mechanism.
 
-A **Sink** is an [_addressable_](./interfaces.md#addressable) resource that
-takes responsibility for the event. A Sink could be a consumer of events, or
-middleware. A Sink MUST be able to receive CloudEvents over HTTP and HTTPS.
+## Event Delivery
 
-A **Sink** MAY be [_callable_](./interfaces.md#callable) resource that
-represents an Addressable endpoint which receives an event as input and
-optionally returns an event to forward downstream.
+To provide simpler support for event sources which might be translating events
+from existing systems, some data plane requirements for senders are relaxed in
+the general case. In the case of Knative Eventing provided resources (Channels
+and Brokers) which implement these roles, requirements are increased from
+SHOULD to MUST. These cases are called out as they occur.
 
-Almost every component in Knative Eventing may be a Sink providing
-composability.
+### Minimum supported protocol
 
-Every Sink MUST support HTTP Protocol Binding for CloudEvents
-[version 1.0](https://github.com/cloudevents/spec/blob/v1.0/http-protocol-binding.md)
+All senders and recipients MUST support the CloudEvents 1.0 protocol and the
```

Comment:
Usually senders only support either binary or structured, and it's a static choice made by the sender implementor.

Reply:
Since sender speaks first, they get to choose. Let me clarify.

Reply:
Not in terms of the HTTP transaction; there's a clearly defined "sender" and "recipient" at the HTTP layer. With that said, I added a sentence clarifying that senders that do not accept replies MAY implement only one content mode.